app/routes.py

```diff
@@ -0,0 +1,279 @@
+        if "phone" not in request_body:
+            return {"details": "Request body must include phone."}, 400
+        else:
+            new_customer = Customer(
+                name= request_body["name"],
+                phone=request_body["phone"],
+                postal_code =request_body["postal_code"])
+
+            db.session.add(new_customer)
+            db.session.commit()
+            new_customer_response = {
+                    "id": new_customer.id,
+                    "name" : new_customer.name,
+                    "phone": new_customer.phone,
+                    "registered_at": new_customer.registered_at,
+                    "postal_code": new_customer.postal_code
+                    }
+            return jsonify (new_customer_response), 201
+
+@customers_bp.route("/<customer_id>", methods=["GET", "DELETE", "PUT"])
+def handle_customer(customer_id):
+    if customer_id.isnumeric() != True:
+        return("Invalid Request"), 400
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+    elif request.method == "GET":
+        return {
+            "id": customer.id,
+            "name" : customer.name,
+            "registered_at" : customer.registered_at,
+            "phone" : customer.phone,
+            "postal_code" : customer.postal_code
+        }, 200
+    elif request.method == "DELETE":
+        db.session.delete(customer)
+        db.session.commit()
+
+        return jsonify({"id": customer.id}), 200
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        if "name" not in form_data:
+            return {"details" : "Invalid data"}, 400 
+    
+        customer.name = form_data["name"]
+        customer.phone = form_data["phone"]
+        customer.postal_code = form_data["postal_code"]
+
+        db.session.commit()
+        return {"name" : customer.name, "phone" : customer.phone, "postal_code": customer.postal_code}, 200
+
+# Wave 2
+def due_date():
+    due_date =datetime.today() + timedelta(days=7)
+    return due_date
+    
+@rentals_bp.route("/check-out", methods=["POST"])
+def handle_checkout():
+    request_body = request.get_json()
+
+    if "customer_id" not in request_body:
+        return {"details": "Request body must include customer_id."}, 400
+    if "video_id" not in request_body:
+        return {"details": "Request body must include video_id."}, 400
+    video_id = request_body["video_id"]
+    video = Video.query.get(video_id)
+    if video is None:
+        return {"message": f"Video {video_id} was not found"}, 404
+        
+    customer_id = request_body["customer_id"]
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return {"message": f"Customer {customer_id} was not found"}, 404
+    movie_due = due_date()
+    if video.total_inventory - Rental.query.filter_by(video_id=video_id).count() == 0:
```

Comment:
another suggestion would be to add `video_checkout_count = db.Column(db.Integer, default=0 `to your Customer model then add to the count and subtract from the inventory count for videos. Then you could do something like
```customer.videos_checked_out_count += 1
    video.available_inventory -= 1
```